### PR TITLE
add checks for DryMode in unit tests

### PR DESCRIPTION
**Pull request by ianvernon**
Wrap certain conditions / operations around `DryMode` being enabled to reduce
the amount of warnings during the unit tests for operations revolving around
filesystem interactions, BPF filesystem being mounted, etc.

Signed-off by: Ian Vernon <ian@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7384)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

File: pkg/kvstore/store/store.go [hunk 1]
```diff
@@ -459,7 +459,10 @@ func (s *SharedStore) watcher(listDone chan bool) {
 
 			case kvstore.EventTypeDelete:
 				if localKey := s.lookupLocalKey(keyName); localKey != nil {
-					logger.Warning("Received delete event for local key. Re-creating the key in the kvstore")
+					if !option.Config.DryMode {
```
Comment (tgraf) [inline pkg/kvstore/store/store.go]: Why is this needed?

**Review by tgraf — CHANGES_REQUESTED** [review]

File: pkg/kvstore/store/store.go (hunk 1 above)
Comment (ianvernon) [reply pkg/kvstore/store/store.go]: There are a ton of these logs that appear during the unit tests - do you think this is something to look into as a potential bug vs. a dry mode problem? The tests still pass, so I figured that it was simply a noisy-log problem relative to how we run the unit tests.

Comment (ianvernon) [reply pkg/kvstore/store/store.go]: Removed for now.

**Comment by ianvernon** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]
